src/vs/workbench/api/common/extHostCommands.ts

```diff
@@ -133,7 +146,9 @@ export class ExtHostCommands implements ExtHostCommandsShape {
 				}
 			});
 
-			return this._proxy.$executeCommand<T>(id, args).then(result => revive(result, 0));
+			return this._proxy.$executeCommand<T>(id, args).then(result => {
+				return revive(result, 0);
+			});
```

Comment:
Unnecessary change?

Reply:
The `executeCommand` function checks if that command itself is contributed by an extension. In that case the execution stays within this process and you need to manually create/fire an event

Reply:
Fixing formatting.

The contributed commands are actually passing through the `commandService`, so events are firing for them.

Reply:
Not those: https://github.com/Microsoft/vscode/blob/23fd99b493979a5b3778dc239d7eea46bf249f16/src/vs/workbench/api/common/extHostCommands.ts#L116

Try with `vscode.commands.executeCommand` and call a command your or another extension contributes.

Reply:
Right right, those are the ones I was referring to.  `extension.helloWorld` seems to already get captured with no changes to the code.  After adding a manual fire after the contributed command is executed in `_executeContributedCommand`, it now fires off twice.  Not 100% sure how, to be honest.

Reply:
It matters how you invoke the command. User gestures like the command-palette or keybindings are happening via the main side should be covered but just calling `executeCommand` (e.g from a timer) is different

Reply:
Added.